api/client/client.go

```diff
@@ -410,8 +418,12 @@ func (c *Client) DownloadManifest(hash string) (*api.Manifest, bool, error) {
 // - a prefix of "dir1/" would return [dir1/dir2/, dir1/file3.txt]
 //
 // where entries ending with "/" are common prefixes.
-func (c *Client) List(hash, prefix, credentials string) (*api.ManifestList, error) {
-	req, err := http.NewRequest(http.MethodGet, c.Gateway+"/bzz-list:/"+hash+"/"+prefix, nil)
+func (c *Client) List(hash, prefix, credentials, publisher string) (*api.ManifestList, error) {
+	uri := c.Gateway + "/bzz-list:/" + hash + "/" + prefix
+	if publisher != "" {
+		uri = uri + "?publisher=" + publisher
+	}
+	req, err := http.NewRequest(http.MethodGet, uri, nil)
```

Comment:
could you please briefly comment on why you needed to make these changes to the `List` func?

Reply:
thank you!

i would like to ask if @zelig has any comments regarding this change.